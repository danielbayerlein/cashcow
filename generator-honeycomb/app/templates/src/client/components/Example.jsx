import React, { PropTypes } from 'react';

/**
 * Example for react on the client
 *
 * @param  {string}   name  display-name
 * @return {element}        react-component
 */
const Example = ({ name }) => (
  <h2>Hi {name}, this is an example for rendering React on the client.</h2>
);

Example.propTypes = {
  name: PropTypes.string,
};

export default Example;
