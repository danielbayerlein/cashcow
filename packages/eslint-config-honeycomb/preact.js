module.exports = {
  extends: [
    'eslint-config-airbnb',
  ].map(require.resolve),
  env: {
    jest: true,
    browser: true,
  },
  rules: {
    'comma-dangle': ['error', {
      arrays: 'always-multiline',
      objects: 'always-multiline',
      imports: 'always-multiline',
      exports: 'always-multiline',
      functions: 'never',
    }],
    'react/jsx-filename-extension': ['error', {
      extensions: ['.js'],
    }],
    'react/prop-types': 'off',
  },
};
