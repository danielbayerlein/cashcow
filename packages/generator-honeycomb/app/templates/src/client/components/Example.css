h2 {
  color: tomato;
  font-size: 2em;
}
