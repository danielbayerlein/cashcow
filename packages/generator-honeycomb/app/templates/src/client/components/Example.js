import React, { PropTypes } from 'react';
import styled from 'styled-components';

const StyledH2 = styled.h2`
  color: tomato;
  font-size: 2em;
`;

/**
 * Example for react on the client
 *
 * @param  {string}   name  display-name
 * @return {element}        react-component
 */
const Example = ({ name }) => (
  <StyledH2>Hi {name}, this is an example for rendering React on the client.</StyledH2>
);

Example.propTypes = {
  name: PropTypes.string,
};

export default Example;
