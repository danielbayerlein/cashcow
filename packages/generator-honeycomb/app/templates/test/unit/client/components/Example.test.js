/** @jsx h */
import { h } from 'preact';
import render from 'preact-render-to-string';
import Example from '../../../../src/client/components/Example';

it('should render correctly', () => {
  const tree = render(<Example name="World" />);
  expect(tree).toMatchSnapshot();
});
