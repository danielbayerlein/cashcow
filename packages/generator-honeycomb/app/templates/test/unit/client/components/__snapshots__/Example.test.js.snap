// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `"<h2>Hi World, this is an example for rendering Preact on the client.</h2>"`;
